src/argon2_node.cpp

```diff
@@ -126,7 +127,7 @@ VerifyWorker::VerifyWorker(std::string&& hash, std::string&& plain):
 
 void VerifyWorker::Execute()
 {
-    auto type = (hash.at(7) == 'd') ? Argon2_d : Argon2_i;
+    auto type = (hash.at(7) == 'd') ? Argon2_d : ( hash.at(8) == 'd' ? Argon2_id : Argon2_i );
```

Comment:
Better to switch to if/elseif/else structure, ternary operator is very ugly here.